### ISSUE TITLE
[Accessibility] Narrator focuses on the Column Headers instead of the first cell when enter DataGridView by switching Form

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#7723]
### .NET version

8.0.100-alpha.1.22430.2

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, it is a regression issue, cannot repro .Net 6.0, for 7.0 version, 

### Issue description

Narrator focuses on the Column Headers instead of the first cell when enter DataGridView by switching Form. If application has one Form with DataGridView, Narrator focuses on the entire DataGridView instead of first cell

Actual Gif:
**2 Forms:**
![image](https://user-images.githubusercontent.com/26474449/188046094-edb8ee01-684d-4af8-b8cc-a8de1804fb6b.png)
![DataGridView_NarratorFocus](https://user-images.githubusercontent.com/26474449/188046113-107a653d-41eb-4aac-aba5-37ce9970c83b.gif)
**1 Form:**
![image](https://user-images.githubusercontent.com/26474449/188048464-23521d3d-a03a-4ead-bb77-8efa80f06cfa.png)

Expected Gif from .Net 6.0:
**2 Forms:**
![image](https://user-images.githubusercontent.com/26474449/188046138-c84b46df-84ba-4d8e-b9cb-1f29019106dc.png)
![DataGridView_NarratorFocus_6 0 gif](https://user-images.githubusercontent.com/26474449/188050894-5d5b2eb1-ed6c-41d4-beba-14c5902dac99.gif)

**1 Form:**
![image](https://user-images.githubusercontent.com/26474449/188048670-2a1c6527-3375-4996-8ff6-1be69892b5ed.png)

### Steps to reproduce

1. Create a Winforms .Net Core application, Form1 with DataGridView added
2. Add several columns in DataGridView
3. Add a Form2 to this project, and add a button to Form2
4. Double click button to generate event with following code
            
```
            Form1 form1 = new Form1();
            form1.ShowDialog();
```
5. Set Form2 as start form, build and run application

or use attached sample application for testing
[DataGridViewSample.zip](https://github.com/dotnet/winforms/files/9474550/DataGridViewSample.zip)

**Comment by merriemcgaw** [comment]
@Olina-Zhang what happens as the user tabs through the application when the column headers are selected?

**Comment by merriemcgaw** [comment]
@vladimir-krestov was there a recent change in the DGV that might be causing this? Can you please find the right person to look into this?

**Comment by Tanya-Solyanik** [comment]
@dmitrii-drobotov  - FYI

**Comment by Olina-Zhang** [comment]
> @Olina-Zhang what happens as the user tabs through the application when the column headers are selected?

it will go to the second cell, skip the first one:
![123](https://user-images.githubusercontent.com/26474449/188777058-743fbe73-9f70-43fd-9de7-48ebb4154c82.gif)

**Comment by MelonWang1** [comment]
Verified this issue in the latest .Net 8.0 SDK build: 8.0.100-alpha.1.22469.15, it is fixed.
**Model window:** Narrator's focusing is in column headers.
![222](https://user-images.githubusercontent.com/94418985/191191466-8b3d5503-9b9a-481b-876c-81e2f76b79f9.gif)

**Not model window:** Narrator's focusing is in first DataGridView cell.
![1form](https://user-images.githubusercontent.com/94418985/191191441-cafc2bee-bd21-4b12-a2a7-b44c95f5716b.gif)

**Comment by MelonWang1** [comment]
@NikitaSemenovAkvelon From your gif, I don't see which keyboard being pressed. In model window, you mean you didn't press the Tab keyboard and when open form designer narrator automatically focusing on column headers, then row header, then first DataGridView cell. If so, above test result is different with your gif. Please see it.

**Comment by NikitaSemenovAkvelon** [comment]
I've retested it from main branch. Below described my steps and a result. It works as expected.
1) Started the application.
2) Pressed enter.
3) Watched how Narrator announced all controls up to the selected cell.
![Rqi3JpCDYA](https://user-images.githubusercontent.com/109065597/191506628-bda12007-755f-4f51-9853-e9275ba064a5.gif)

**Comment by MelonWang1** [comment]
@NikitaSemenovAkvelon Verified this issue in the latest .Net 8.0 SDK build: 8.0.100-alpha.1.22470.29,
In Win11 21H2(OS Build 22000.856), this issue fixed and result same with you.
But in Win10 21H2(OS Build 19044.2006), this issue not fixed and result as above mine attached.
